src/features/auth/model/auth.interceptor.ts

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios';
+import type { AxiosError, AxiosInstance } from 'axios';
+import { cookieManager } from '@shared/lib/cookieManager.ts';
+import { refreshApi } from '../api';
+
+export const setupAuthInterceptors = (Api: AxiosInstance): void => {
+  Api.interceptors.request.use(
+    (config) => {
+      const token = cookieManager.getCookie('accessToken');
+      if (token) {
+        config.headers.Authorization = `Bearer ${token}`;
+      }
+      return config;
+    },
+    (error) => {
+      return Promise.reject(new Error(error as string));
+    }
+  );
+
+  Api.interceptors.response.use(
+    (response) => {
+      return response;
+    },
+    async (error: AxiosError) => {
+      const originalRequest = error.config;
+      if (error.response?.status === 401) {
```

Comment:
```suggestion
      if (error.response?.status === HttpStatusCodes.Unautorized) {
```
No magic numbers. Separate `HttpStatusCodes` to `shared/enums`